### DISCUSSION TITLE
What is source:rnv and how to change it to specific custom version from fork

### DISCUSSION THREAD

**Discussion by ParvinderjitSF** [pavjacko/renative#1535] [Q&A] [answered]
What is the meaning of source:rnv. How easy it would to use the custom version of the react-native ?

**Comment by ElenaDiachenko** [comment, marked as answer]
@ParvinderjitSF 
There is no easy way to change the react-native version because rnv also uses a project template that is configured according to the current react-native version. The term "source: rnv" refers to the location where the current plugin implementation is defined (in this case, it’s `node_modules/@rnv/config-templates/renative.templates.json`).

## Steps for modifying the react-native version:
1. change the version in the package. json;
2. configure the plugin in renative. json;
```
 "plugins": {
        "react-native":{
            "version": "0.74.0",
            "disablePluginTemplateOverrides": true
        }
    },
```
3. use platform build overrides to change files which are modified according to react-native upgrade. ( https://react-native-community.github.io/upgrade-helper/). All files located in  your_project/appConfigs/base/builds/platform  will replace the same files in the generated project inside platformBuilds.

<img width="286" alt="Screenshot 2024-09-23 at 11 35 19" src="https://github.com/user-attachments/assets/add0788e-0e60-4c66-8588-4112f03c1ed7">

 Keep in mind that upgrading to a major version may not work due to possible  breaking changes in cli.
